lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -10,259 +10,333 @@
             'teams/js/models/topic',
             'teams/js/collections/topic',
+            'teams/js/views/team_profile',
             'teams/js/views/teams',
             'teams/js/collections/team',
+            'teams/js/models/team',
             'text!teams/templates/teams_tab.underscore'],
-           function (Backbone, _, gettext, HeaderView, HeaderModel, TabbedView,
-                     TopicsView, TopicModel, TopicCollection, TeamsView, TeamCollection, teamsTemplate) {
-               var ViewWithHeader = Backbone.View.extend({
-                   initialize: function (options) {
-                       this.header = options.header;
-                       this.main = options.main;
-                   },
+        function (Backbone, _, gettext, HeaderView, HeaderModel, TabbedView,
+                  TopicsView, TopicModel, TopicCollection, TeamProfileView, TeamsView, TeamCollection, TeamModel,
+                  teamsTemplate) {
+            var ViewWithHeader = Backbone.View.extend({
+                initialize: function (options) {
+                    this.header = options.header;
+                    this.main = options.main;
+                },
 
-                   render: function () {
-                       this.$el.html(_.template(teamsTemplate));
-                       this.$('p.error').hide();
-                       this.header.setElement(this.$('.teams-header')).render();
-                       this.main.setElement(this.$('.page-content')).render();
-                       return this;
-                   }
-               });
+                render: function () {
+                    this.$el.html(_.template(teamsTemplate));
+                    this.$('p.error').hide();
+                    this.header.setElement(this.$('.teams-header')).render();
+                    this.main.setElement(this.$('.page-content')).render();
+                    return this;
+                }
+            });
+            var TeamTabView = Backbone.View.extend({
+                initialize: function(options) {
+                    var TempTabView, router;
+                    this.courseID = options.courseID;
+                    this.topics = options.topics;
+                    this.topicUrl = options.topicUrl;
+                    this.teamsUrl = options.teamsUrl;
+                    this.maxTeamSize = options.maxTeamSize;
+                    // This slightly tedious approach is necessary
+                    // to use regular expressions within Backbone
+                    // routes, allowing us to capture which tab
+                    // name is being routed to
+                    router = this.router = new Backbone.Router();
+                    _.each([
+                        [':default', _.bind(this.routeNotFound, this)],
+                        ['topics/:topic_id', _.bind(this.browseTopic, this)],
+                        ['teams/:topic_id/:team_id', _.bind(this.browseTeam, this)],
```

Comment:
You are browsing from the list of teams into a single team, from which you can see the discussions, members etc. It is singular to indicate this, as with the previous line's ``browseTopic`` which lets you browse the contents of a single topic. 

We've always had trouble with naming the list view versus the individual view. Having an extra ``s`` to indicate the plural is confusing, but I'm not sure I like the ``ListView`` vs ``DetailView`` convention either. Let's come to an agreement as to how to name this and then I can do the renaming.